frontend/js/src/settings/music-services/details/MusicServices.tsx

```diff
@@ -257,10 +283,135 @@ export default function MusicServices() {
+      if (navidromeIsEditing && permissions.navidrome === "listen") {
+        try {
+          const disconnectResponse = await fetch(
+            `/settings/music-services/navidrome/disconnect/`,
+            {
+              method: "POST",
+              headers: {
+                "Content-Type": "application/json",
+                Authorization: `Token ${currentUser?.auth_token}`,
+              },
+            }
+          );
+        } catch (disconnectError) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            "Failed to disconnect before reconnecting:",
+            disconnectError
+          );
+        }
+      }
+
+      const response = await fetch(
+        `/settings/music-services/navidrome/connect/`,
+        {
+          method: "POST",
+          body: JSON.stringify({
+            host_url: hostUrl,
+            username,
+            password,
+          }),
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Token ${currentUser?.auth_token}`,
+          },
+        }
+      );
+
+      let data;
+      const contentType = response.headers.get("content-type");
+      if (contentType && contentType.includes("application/json")) {
+        data = await response.json();
+      } else {
+        throw Error("Server returned non-JSON response");
+      }
+
+      if (response.ok) {
+        if (navidromeAuth) {
+          navidromeAuth.instance_url = hostUrl;
+          navidromeAuth.username = username;
+        }
+
+        // Update edit values state
+        setNavidromeEditValues({
+          hostUrl,
+          username,
+        });
+
+        // Exit edit mode if we were editing
+        setNavidromeIsEditing(false);
+
+        toast.success(
+          <ToastMsg
+            title="Success"
+            message="Successfully connected to Navidrome!"
+          />
+        );
+        setPermissions((prev) => ({ ...prev, navidrome: "listen" }));
+      } else if (data?.error) {
+        throw Error(data.error);
+      } else {
+        throw Error(`Server error: ${response.status} ${response.statusText}`);
+      }
+    } catch (error) {
+      toast.error(
+        <ToastMsg
+          title="Failed to connect to Navidrome"
+          message={error.toString()}
+        />
+      );
+    }
+  };
+
+  const handleNavidromeEditToggle = () => {
+    if (navidromeIsEditing) {
+      // If we're saving, submit the form
+      const form = document.getElementById("navidrome-form") as HTMLFormElement;
+      if (form) {
+        form.requestSubmit();
+      }
+    } else {
+      // Start editing
+      setNavidromeIsEditing(true);
+    }
+  };
+
+  React.useEffect(() => {
+    setNavidromeEditValues({
+      hostUrl: navidromeAuth?.instance_url || "",
+      username: navidromeAuth?.username || "",
+    });
+  }, [navidromeAuth?.instance_url, navidromeAuth?.username]);
```

Comment:
The only place I saw where you set the instanceUrl and username are at line 353-355, and you already call setNavidromeEditValues right after that.
Is this effect hook necessary?